clustered/server/src/main/java/org/ehcache/clustered/server/offheap/OffHeapChainStorageEngine.java

```diff
@@ -201,14 +205,7 @@ public int length() {
 
   }
 
-  private static class PrimordialChain implements InternalChain {
-
-    private final ByteBuffer element;
-
-    public PrimordialChain(ByteBuffer element) {
-      this.element = element;
-    }
-
+  private static abstract class Rudimentary implements InternalChain {
```

Comment:
Rudimentary is an adjective. So it is weird for a class name. But I have now clue of the intend to help find a better word. Can you javadoc it to explain what it is?

Reply:
I will change it. I was also not completely happy with that. Will RudimentaryObject sound better?

Reply:
It's a chain. Maybe the name should have chain in it somewhere?

Reply:
I RudimentaryChain and RudimentaryElement extending from Rudimentary 
  Does GenesisChain,  GenesisComplexChain and GenesisSimpleChain sound better?